core/src/Kokkos_View.hpp

```diff
@@ -132,6 +132,66 @@ void runtime_check_rank_host(const size_t dyn_rank, const bool is_void_spec,
 }
 #endif
 
+template <typename Traits, typename Layout>
+KOKKOS_INLINE_FUNCTION void runtime_check_rank(
```

Comment:
do the same as in the above: instead of templating on Traits: add a parameter for `dyn_rank` and `is_void_spec`. Essentially make the interface like the `runtime_check_rank_host` just that you get a Layout instead of integers.